src/base/utils/net.cpp

```diff
@@ -38,5 +38,27 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isLoopbackAddress(const QHostAddress &addr)
+        {
+            return addr == QHostAddress::LocalHost
+                    || addr == QHostAddress::LocalHostIPv6
+                    || addr == QHostAddress(QLatin1String("::ffff:127.0.0.1"));
+        }
+
+        bool isLANAddress(const QHostAddress &addr)
+        {
+            // ipv4
+            if (addr.isInSubnet(QHostAddress(QLatin1String("0.0.0.0")), 0)) {
```

Comment:
just stumble upon this, there is a simpler check for this:
```c++
if (addr.protocol() == QAbstractSocket::IPv4Protocol)
```
http://doc.qt.io/qt-5/qhostaddress.html#protocol

pls test if it works